crates/backend/src/ast.rs

```diff
@@ -261,9 +259,9 @@ pub struct Enum {
 
 #[cfg_attr(feature = "extra-traits", derive(Debug, PartialEq, Eq))]
 #[derive(Clone)]
-pub struct Variant {
+pub struct Variant<Val = u32> {
```

Comment:
A type parameter here seems a bit like overkill, is this used in multiple locations to warrant that?

Reply:
It's used by `ast::ImportEnum` where `Val = String`, but this isn't actually used for preserving attributes. As mentioned, I made this change so that TypeScript bindings for string enums could have the docs, but I wasn't sure if this would be considered useful/the best way to do this.

Reply:
Would it be possible to either update this structure to store a `u32` or `String`? Is it still needed in two different contexts at the same time?